### ISSUE TITLE
Assert_failure app/functoria_graph.ml:248:32

### ISSUE THREAD

**Issue by talex5** [mirage/functoria#70]
In a directory containing a single file, `config.ml`, with:

```
open Mirage

let foo =
  foreign "Unikernel.Main" (stackv4 @-> job)

let stack =
  if_impl Key.is_xen
    (generic_stackv4 default_console (netif "0"))
    (socket_stackv4 default_console [Ipaddr.V4.any])

let () =
  register "dns" [foo $ stack]
```

I get:

```
$ mirage configure --no-opam

$ mirage configure --no-opam --unix
Fatal error: exception "Assert_failure app/functoria_graph.ml:248:32"

$ mirage configure --no-opam -t unix
Fatal error: exception "Assert_failure app/functoria_graph.ml:248:32"
```

(also happens without `--no-opam`, just more slowly)

```
$ opam list
[...]
functoria                1.1.1  A DSL to organize functor applications.
mirage                   2.9.1  The MirageOS library operating system
```

**Comment by hannesm** [comment]
I run into this issue as well, @Drup any chance you can have a look at this?

**Comment by yomimono** [comment]
I can't reproduce this with the current `mirage-dev` set of stuff.  (I had to remove `default_console`, FWIW.)

**Comment by hannesm** [comment]
wasn't this fixed by #83?

**Comment by Drup** [comment]
Yes it was.
